src/core/operators/base_op.py

```
@@ -0,0 +1,124 @@
+        
+        Returns:
+            List[Any]: List of tool objects for the agent
+        """
+        mcp_zoo = MCPZoo(mcp_names=self.mcps)
+        if not self.mcps:
+            return []
+
+        # Ensure we have an MCPZoo. If none provided, attempt a safe default for convenience.
+        try:
+            tools = await mcp_zoo.get_tools(self.mcps)
+            return tools
```

Ruff (0.12.2)

Consider moving this statement to an `else` block

(TRY300)